From 82ba232fa8a058903385184c07bdba271604fd3e Mon Sep 17 00:00:00 2001
From: Lichtenshtein <altruumann@protonmail.com>
Date: Sun, 7 Dec 2025 02:11:19 +0300
Subject: [PATCH] Change default palette to SPECTRUM

---
diff --git a/src/spek-palette.h b/src/spek-palette.h
index 27434de..4772e0e 100644
--- a/src/spek-palette.h
+++ b/src/spek-palette.h
@@ -7,7 +7,7 @@ enum palette {
     PALETTE_SOX,
     PALETTE_MONO,
     PALETTE_COUNT,
-    PALETTE_DEFAULT = PALETTE_SOX,
+    PALETTE_DEFAULT = PALETTE_SPECTRUM,
 };
 
 uint32_t spek_palette(enum palette palette, double level);
