diff --git a/src/ffmpeg.mk b/src/ffmpeg.mk
index b486d859..d1d7013d 100644
--- a/src/ffmpeg.mk
+++ b/src/ffmpeg.mk
@@ -8,9 +8,9 @@ $(PKG)_CHECKSUM := 733984395e0dbbe5c046abda2dc49a5544e7e0e1e2366bba849222ae9e3a0
 $(PKG)_SUBDIR   := $(PKG)-$($(PKG)_VERSION)
 $(PKG)_FILE     := $(PKG)-$($(PKG)_VERSION).tar.xz
 $(PKG)_URL      := https://ffmpeg.org/releases/$($(PKG)_FILE)
-$(PKG)_DEPS     := cc bzip2 gnutls lame libass libbluray libbs2b libcaca \
-                   libvpx opencore-amr opus sdl2 speex theora vidstab \
-                   vo-amrwbenc vorbis x264 x265 xvidcore yasm zlib
+$(PKG)_DEPS     := cc bzip2 lame libbluray \
+                   libvpx opencore-amr opus sdl2 speex \
+                   vorbis yasm zlib
 
 # DO NOT ADD fdk-aac OR openssl SUPPORT.
 # Although they are free softwares, their licenses are not compatible with
@@ -42,25 +42,26 @@ define $(PKG)_BUILD
         --disable-doc \
         --enable-gpl \
         --enable-version3 \
+        --disable-programs \
+        --disable-avdevice \
+        --disable-swresample \
+        --disable-swscale \
+        --disable-postproc \
+        --disable-avfilter \
+        --disable-encoders \
+        --disable-muxers \
+        --disable-devices \
+        --disable-filters \
         --extra-libs='-mconsole' \
-        --enable-gnutls \
-        --enable-libass \
         --enable-libbluray \
-        --enable-libbs2b \
-        --enable-libcaca \
         --enable-libmp3lame \
         --enable-libopencore-amrnb \
         --enable-libopencore-amrwb \
         --enable-libopus \
         --enable-libspeex \
-        --enable-libtheora \
-        --enable-libvidstab \
-        --enable-libvo-amrwbenc \
         --enable-libvorbis \
         --enable-libvpx \
-        --enable-libx264 \
-        --enable-libx265 \
-        --enable-libxvid \
+        --enable-small
         --extra-ldflags="-fstack-protector" \
         $($(PKG)_CONFIGURE_OPTS)
     $(MAKE) -C '$(BUILD_DIR)' -j '$(JOBS)'
