diff --git a/src/ffmpeg.mk b/src/ffmpeg.mk
index b486d859..d1d7013d 100644
--- a/src/ffmpeg.mk
+++ b/src/ffmpeg.mk
@@ -42,25 +42,27 @@ define $(PKG)_BUILD
         --disable-doc \
         --enable-gpl \
         --enable-version3 \
+        --disable-programs \
+        --disable-avdevice \
+        --disable-swresample \
+        --disable-swscale \
+        --disable-postproc \
+        --disable-avfilter \
+        --disable-encoders \
+        --disable-muxers \
+        --disable-devices \
+        --disable-filters \
         --extra-libs='-mconsole' \
-        --enable-gnutls \
-        --enable-libass \
         --enable-libbluray \
-        --enable-libbs2b \
-        --enable-libcaca \
         --enable-libmp3lame \
         --enable-libopencore-amrnb \
         --enable-libopencore-amrwb \
         --enable-libopus \
         --enable-libspeex \
-        --enable-libtheora \
-        --enable-libvidstab \
-        --enable-libvo-amrwbenc \
         --enable-libvorbis \
         --enable-libvpx \
-        --enable-libx264 \
-        --enable-libx265 \
-        --enable-libxvid \
+        --enable-small
+        --extra-cflags="-march=native -Ofast -pipe"
         --extra-ldflags="-fstack-protector" \
         $($(PKG)_CONFIGURE_OPTS)
     $(MAKE) -C '$(BUILD_DIR)' -j '$(JOBS)'
