From 78e29116caa988dcfadff1b08ab8fb3f82dc9963 Mon Sep 17 00:00:00 2001
From: MikeWang000000 <mikewang000000@gmail.com>
Date: Fri, 24 Jun 2022 16:41:37 +0800
Subject: [PATCH] macOS: Cmd-W to close the window

---
 src/spek-spectrogram.cc | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/src/spek-spectrogram.cc b/src/spek-spectrogram.cc
index aceb170..b2ba763 100644
--- a/src/spek-spectrogram.cc
+++ b/src/spek-spectrogram.cc
@@ -93,7 +93,17 @@ void SpekSpectrogram::save(const wxString& path)
 
 void SpekSpectrogram::on_char(wxKeyEvent& evt)
 {
-    switch (evt.GetKeyCode()) {
+    int key_modifier = evt.GetModifiers();
+    int key_code = evt.GetKeyCode();
+
+#ifdef OS_OSX
+    // Using Command-W to close the window in macOS
+    if (key_modifier == wxMOD_CONTROL && key_code == 'W') {
+        this->GetParent()->Close(true);
+        return;
+    }
+#endif
+    switch (key_code) {
     case 'f':
         this->window_function = (enum window_function) ((this->window_function + 1) % WINDOW_COUNT);
         break;
