From e98b11294d929a5f62b5df7354743fcbb2729fb0 Mon Sep 17 00:00:00 2001
From: Alexander Kojevnikov <alexander@kojevnikov.com>
Date: Sun, 8 Jan 2023 18:39:43 -0800
Subject: [PATCH] Fix decoding for m4a and ogg

---
 src/spek-audio.cc | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/spek-audio.cc b/src/spek-audio.cc
index 48f2aae..c790045 100644
--- a/src/spek-audio.cc
+++ b/src/spek-audio.cc
@@ -73,7 +73,7 @@ std::unique_ptr<AudioFile> Audio::open(const std::string& file_name, int stream)
     AudioError error = AudioError::OK;
 
     AVFormatContext *format_context = nullptr;
-    if (avformat_open_input(&format_context, file_name.c_str(), nullptr, nullptr) != 0) {
+    if (avformat_open_input(&format_context, file_name.c_str(), nullptr, nullptr) < 0) {
         error = AudioError::CANNOT_OPEN_FILE;
     }
 
@@ -293,8 +293,10 @@ int AudioFileImpl::read()
             this->packet->data += len;
             this->packet->size -= len;
             this->offset += len;
-            // We have data, return it and come back for more later.
             int samples = this->frame->nb_samples;
+            // Occasionally the frame has no samples, move on to the next one.
+            if (samples == 0) continue;
+            // We have data, return it and come back for more later.
             if (samples > this->buffer_len) {
                 this->buffer = static_cast<float*>(
                     av_realloc(this->buffer, samples * sizeof(float))
