From 214b67d0718e7af6b371de2296273f8b15a67193 Mon Sep 17 00:00:00 2001
From: MikeWang000000 <mikewang000000@gmail.com>
Date: Wed, 25 Jan 2023 22:41:08 +0800
Subject: [PATCH] Fix unit tests

---
 tests/test-audio.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/test-audio.cc b/tests/test-audio.cc
index c0d6e59..d2b002e 100644
--- a/tests/test-audio.cc
+++ b/tests/test-audio.cc
@@ -100,7 +100,7 @@ void test_audio()
         {"2ch-44100Hz.ac3",
             {AudioError::OK, "ATSC A/52", 192000, 44100, 0, 2, AC3_T, 4608}},
         {"2ch-44100Hz-std.mpc",
-            {AudioError::OK, "Musepack", 0, 44100, 0, 2, 0.0, 6912}},
+            {AudioError::OK, "Musepack", 0, 44100, 0, 2, 0.0, 5760}},
         {"2ch-44100Hz-v1.wma",
             {AudioError::OK, "Windows Media Audio 1", 128000, 44100, 0, 2, 0.138, 4 * 1024}},
         {"2ch-44100Hz-v2.wma",
