From 1bdddb0cbcba7e38e5cbe19bf693a373002a72c4 Mon Sep 17 00:00:00 2001
From: Sami Farin <hvtaifwkbgefbaei@gmail.com>
Date: Fri, 17 Jan 2025 21:41:21 +0200
Subject: [PATCH] fix channel processing for non-planar data

---
 src/spek-audio.cc | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/spek-audio.cc b/src/spek-audio.cc
index 5bb18b6..53a6ade 100644
--- a/src/spek-audio.cc
+++ b/src/spek-audio.cc
@@ -1,3 +1,4 @@
+#include <wx/log.h>
 #include <assert.h>
 
 extern "C" {
@@ -313,7 +314,7 @@ int AudioFileImpl::read()
                     offset = sample;
                 } else {
                     data = this->frame->data[0];
-                    offset = sample * this->channels;
+                    offset = sample * this->channels + this->channel;
                 }
                 float value;
                 switch (format) {
